src/shaders/particles.glsl

```diff
@@ -79,6 +79,12 @@ vec2 update(vec2 pos) {
     vec2 velocity = mix(u_wind_min, u_wind_max, lookup_wind(wind_tex_pos));
     float speed_t = length(velocity) / length(u_wind_max);
 
+    // lookup wind mask in b channel
```

Comment:
This is the  main point of this  version. I added a  "blue" mask channel which allows  you to mask certain areas (for example land in ocean models).